Creating OCP performance guide (#564)
RedHatInsights/red-hat-ansible-automation-platform-documentation: downstream/modules/platform/proc-set-custom-pod-timeout.adoc

[id="proc-set-custom-pod-timeout"]

= Custom pod timeouts

> You can set `awx_container_group_pod_pending_timeout` higher if long running jobs are being canceled. 

Reviewer: ```suggestion
A container group job in Controller is transitioned to the `running` state just before we submit the pod to the kubernetes API. It is then expected by Controller that the pod will enter the "Running" state within `awx_container_group_pod_pending_timeout` seconds. You can set `awx_container_group_pod_pending_timeout` higher value if it is desired that Controller wait for longer before cancelling jobs that fail to enter into the "Running" state.
```

may want to link out to the openshift docs about pod scheduling. https://access.redhat.com/documentation/en-us/openshift_container_platform/4.11/html/nodes/controlling-pod-placement-onto-nodes-scheduling

[id="proc-set-custom-pod-timeout"]

= Custom pod timeouts

A container group job in {ControllerName} will transition to the `running` state just before you submit the pod to the kubernetes API. 
{ControllerNameStart} then expects the pod to enter the `Running` state before `awx_container_group_pod_pending_timeout` seconds has elapsed. 
You can set `awx_container_group_pod_pending_timeout` to a higher value if you want {ControllerName} to wait for longer before cancelling jobs that fail to enter the `Running` state.

> For further information about pod scheduling, see link:https://access.redhat.com/documentation/en-us/openshift_container_platform/4.11/html/nodes/controlling-pod-placement-onto-nodes-scheduling[Controlling pod placement onto nodes].

Reviewer 2: ```suggestion
For further information about pod scheduling, see link:https://access.redhat.com/documentation/en-us/openshift_container_platform/{OCPLatest}/html/nodes/controlling-pod-placement-onto-nodes-scheduling[Controlling pod placement onto nodes].
```

Attribute for latest version of OpenShift

[id="proc-set-custom-pod-timeout"]

.Extra settings
With `extra_settings`, you can pass multiple custom settings using the awx-operator. 
The parameter `extra_settings` is appended to `/etc/tower/settings.py` and can be an alternative to the `extra_volumes` parameter.

[cols="20%,20%,20%",options="header"]
|====
| Name | Description |Default
| `extra_settings` | Extra settings | ‘’
|====

.Example configuration of `extra_settings` parameter

[options="nowrap" subs="+quotes,attributes"]
----
 spec:
    extra_settings:
      - setting: MAX_PAGE_SIZE
        value: "500"

      - setting: AUTH_LDAP_BIND_DN
        value: "cn=admin,dc=example,dc=com"

      - setting: SYSTEM_TASK_ABS_MEM
        value: "500"
----

= Custom pod timeouts

> A container group job in {ControllerName} will transition to the `running` state just before you submit the pod to the kubernetes API. 

Reviewer 2: ```suggestion
A container group job in {ControllerName} transitions to the `running` state just before you submit the pod to the kubernetes API. 
```

{ControllerNameStart} then expects the pod to enter the `Running` state before `AWX_CONTAINER_GROUP_POD_PENDING_TIMEOUT` seconds has elapsed. 
You can set `AWX_CONTAINER_GROUP_POD_PENDING_TIMEOUT` to a higher value if you want {ControllerName} to wait for longer before cancelling jobs that fail to enter the `Running` state.
`AWX_CONTAINER_GROUP_POD_PENDING_TIMEOUT` is how long {ControllerName} waits from creation of a pod until the Ansible work begins in the pod. 
You can also extend the time if the pod cannot be scheduled becaus of resource constraints. 

> You can do this using `extra_settings` on the {Controllername}specification.

Reviewer 2: ```suggestion
You can do this using `extra_settings` on the {Controllername} specification.
```

The default value is two hours. 

> This is used if you are consistently launching many more jobs than kubernetes can schedule, and jobs are spending periods longer than `AWX_CONTAINER_GROUP_POD_PENDING_TIMEOUT` in pending. 

Reviewer 2: ```suggestion
This is used if you are consistently launching many more jobs than Kubernetes can schedule, and jobs are spending periods longer than `AWX_CONTAINER_GROUP_POD_PENDING_TIMEOUT` in pending. 
```

Jobs are not launched until control capacity is available. 

> If many more jobs are being launched than the container group has capacity to run, consider scaling up your kubernetes worker nodes.

Reviewer 2: ```suggestion
If many more jobs are being launched than the container group has capacity to run, consider scaling up your Kubernetes worker nodes.
```
